compose.js

```diff
@@ -69,40 +121,50 @@ function createStamp(descriptor, composeFunction) {
  * @returns {object} Returns the dstDescriptor argument.
  */
 function mergeComposable(dstDescriptor, srcComposable) {
-  const srcDescriptor = (srcComposable && srcComposable.compose) || srcComposable;
+  // Check if it's a stamp or something else.
+  const srcDescriptor = isStamp(srcComposable) ? srcComposable.compose : srcComposable;
+  // Ignore everything but things we can merge.
   if (!isDescriptor(srcDescriptor)) return dstDescriptor;
 
   const combineProperty = (propName, action) => {
-    if (!isObject(srcDescriptor[propName])) return;
-    if (!isObject(dstDescriptor[propName])) dstDescriptor[propName] = {};
+    // Do not create destination properties if there is no need.
```

Comment:
Out of curiosity, what's the advantage to using a lambda here instead of a function expression or statement?